### ISSUE TITLE
Select component should use focusable elements

### ISSUE THREAD

**Issue by jgiardino** [patternfly/patternfly-next#1280]
The elements used in the Select component that are interactive should use native html elements so that they can receive keyboard focus.

- In examples where clicking anywhere on the toggle has the same behavior, the whole toggle wrapping element should be a button. For example in the first example, a `<div>` is used for for `.pf-c-select__toggle`. instead of `<button>`.
- In examples where the toggle includes separate interactive elements with distinct behaviors, check that each element uses an element type that is appropriate for the behavior associated with it. For example, if clicking on the left places focus in an input, but clicking on the caret on the right displays a menu, then the caret icon should be wrapped in a button. For example, **Select - Single with Typeahead (Expanded and Selected)** uses`<i class="fas fa-caret-down pf-c-select__toggle-arrow" aria-hidden="true"></i>` without a button element. If this should be clickable, it should be wrapped with a button.

As part of this issue, any updates to the html should be reflected in the documentation.
